fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -1384,4 +1381,75 @@ public void updateLoanApplicationAttributes(JsonCommand command, Loan loan, Map<
             loanProductRelatedDetail.setEqualAmortization(newValue);
         }
     }
+
+    public Pair<Loan, Map<String, Object>> assembleLoanApproval(AppUser currentUser, JsonCommand command, Long loanId) {
+        final JsonArray disbursementDataArray = command.arrayOfParameterNamed(LoanApiConstants.disbursementDataParameterName);
+        final Loan loan = this.loanRepositoryWrapper.findOneWithNotFoundDetection(loanId, true);
+
+        final LoanStatus newStatus = defaultLoanLifecycleStateMachine.dryTransition(LoanEvent.LOAN_APPROVED, loan);
```

Comment:
I believe this check was moved into the validator, right? If yes, i think this can be removed.